packages/web/src/builders/build/build.impl.ts

```diff
@@ -36,40 +48,98 @@ export function run(
   options: WebBuildBuilderOptions,
   context: BuilderContext
 ): Observable<BuildResult> {
-  return from(getSourceRoot(context)).pipe(
+  const host = new NodeJsSyncHost();
+  return from(getSourceRoot(context, host)).pipe(
     map(sourceRoot => {
       options = normalizeWebBuildOptions(
         options,
         context.workspaceRoot,
         sourceRoot
       );
-      let config = getWebConfig(
-        context.workspaceRoot,
-        sourceRoot,
-        options,
-        context.logger
-      );
-      if (options.webpackConfig) {
-        config = require(options.webpackConfig)(config, {
+      let configs = [
+        getWebConfig(
+          context.workspaceRoot,
+          sourceRoot,
           options,
-          configuration: context.target.configuration
-        });
+          context.logger,
+          options.differentialLoading ? ScriptTarget.ES2015 : null
+        )
+      ];
+      if (options.differentialLoading) {
+        configs.push(
+          getWebConfig(
+            context.workspaceRoot,
+            sourceRoot,
+            options,
+            context.logger,
+            ScriptTarget.ES5
+          )
+        );
       }
-      return config;
+      if (options.webpackConfig) {
+        configs = configs.map(config =>
+          require(options.webpackConfig)(config, {
+            options,
+            configuration: context.target.configuration
+          })
+        );
+      }
+      return configs;
     }),
-    concatMap(config =>
-      runWebpack(config, context, {
+    concatMap(configs => {
+      const runWebpackOptions = {
         logging: stats => {
-          if (options.statsJson) {
-            writeFileSync(
-              resolve(context.workspaceRoot, options.outputPath, 'stats.json'),
-              JSON.stringify(stats.toJson(), null, 2)
-            );
-          }
-
           context.logger.info(stats.toString());
         }
-      })
-    )
+      };
+      return from(configs).pipe(
+        mergeScan(
```

Comment:
would not forkJin work better here? I might be misunderstanding what you are trying to do